scipy/signal/bsplines.py

```diff
@@ -322,7 +322,7 @@ def cspline1d_eval(cj, newx, dx=1.0, x0=0):
 
     """
     newx = (asarray(newx) - x0) / float(dx)
-    res = zeros_like(newx)
+    res = zeros(newx.shape,dtype=cj.dtype)
```

Comment:
Those are both good suggestions. It appears that zeros_like is a better choice since zeros here because it will attempt to keep the same memory layout as the original which could be more efficient. I made both changes on my fork.